src/main/java/com/synopsys/integration/detect/lifecycle/run/RunManager.java

```diff
@@ -141,6 +157,8 @@ public RunResult run(ProductRunData productRunData) throws DetectUserFriendlyExc
         BdioCodeLocationCreator bdioCodeLocationCreator = detectContext.getBean(BdioCodeLocationCreator.class);
         DetectInfo detectInfo = detectContext.getBean(DetectInfo.class);
         DetectDetectableFactory detectDetectableFactory = detectContext.getBean(DetectDetectableFactory.class);
+        blackDuckServicesFactory = productRunData.getBlackDuckRunData().getBlackDuckServicesFactory().orElse(null); // TODO - what is best way to provide blackduck service to populateEnabledTools?
+        enabledTools = fetchEnabledTools();
```

Comment:
So we'd fetch the enabled tools in BlackDuckConnectivity checker, then pass that list to BlackDuckRunData, then we somehow would pass that information to runOptions so it could determine which tool filter to use?

Reply:
I think I've changed my mind because the additional log seems valuable. I do think it should be implemented more similarly to the ToolFilter. And when we are offline, we should disable filtering (filter subclass or otherwise). Maybe DetectToolFeatureFilter with isEnabled(tool).